### DISCUSSION TITLE
Is ImageSharp affected by CVE-2023-5129?

### DISCUSSION THREAD

**Discussion by SigmundurMorkore** [SixLabors/ImageSharp#2538] [Q&A] [answered]
Is ImageSharp affected by CVE-2023-5129?

> With a specially crafted WebP lossless file, libwebp may write data out of bounds to the heap. The ReadHuffmanCodes() function allocates the HuffmanCode buffer with a size that comes from an array of precomputed sizes: kTableSize. The color_cache_bits value defines which size to use. The kTableSize array only takes into account sizes for 8-bit first-level table lookups but not second-level table lookups. libwebp allows codes that are up to 15-bit (MAX_ALLOWED_CODE_LENGTH). When BuildHuffmanTable() attempts to fill the second-level tables it may write data out-of-bounds. The OOB write to the undersized array happens in ReplicateValue. 

- https://www.cve.org/CVERecord?id=CVE-2023-5129

**Comment by antonfirsov** [comment, marked as answer]
I see no unsafe code [in our version of that method](https://github.com/SixLabors/ImageSharp/blob/63d1d2c10b7aaee34efd85b6009e29e82f67b839/src/ImageSharp/Formats/Webp/Lossless/WebpLosslessDecoder.cs#L363-L496) or in other related code, so likely there are no buffer overruns and the worst thing that would happen is an `IndexOutOfRangeException`. It would make it much easier to reproduce/verify this if there was a test image available somewhere.
/cc @brianpopow
